### PR TITLE
Add mTLS Support

### PR DESCRIPTION
**Pull request by SirNexus**
Added mTLS support for 0.7.1. Added configuration for TLS when ControlPlaneAuthPolicy is set to "MUTUAL_TLS".

### CODE REVIEW

File: istio/agent/pilot/client.go [hunk 1]
```diff
@@ -35,7 +37,57 @@ type ProxyConfig struct {
 // NewClient creates a new Client.
 func NewClient(endpoint string, httpClient *http.Client, serviceNode string, serviceCluster string, podIP string,
 	collectorAddress string,collectorTopic string) *Client {
-	return &Client{fmt.Sprintf("http://%v", endpoint), httpClient, serviceNode, serviceCluster, podIP,collectorAddress,collectorTopic}
+	return &Client{endpoint, httpClient, serviceNode, serviceCluster, podIP,collectorAddress,collectorTopic}
+}
+
+// ConfigureAuth sets up appropriate transport config for client
+func ConfigureAuth(authPolicy string) *http.Client {
```
Comment (pleshakov) [inline istio/agent/pilot/client.go]: move `ConfigureAuth` and `GetEndpoint` to main.go

File: istio/agent/pilot/client.go [hunk 2]
```diff
@@ -35,7 +37,57 @@ type ProxyConfig struct {
...
+				TLSClientConfig: &tls.Config {
+					Certificates: []tls.Certificate{mTLSConfig},
+					RootCAs: rootCA,
+				},
+			}
+			return &http.Client{Transport: tr}
+		case "NONE":
+			return &http.Client{}
+		default:
+			panic(fmt.Sprintf("Unknown auth policy: %v\n", authPolicy))
```
Comment (pleshakov) [inline istio/agent/pilot/client.go]: Why not glog.Fatal

File: istio/agent/pilot/client.go [hunk 3]
```diff
@@ -224,7 +276,8 @@ func (c *Client) GetConfig() ProxyConfig {
 
 	listeners, err := c.getListeners()
 	if err != nil {
-		glog.Fatalf("Error getting listeners: %v", err)
+		glog.Infof("Error getting listeners: %v", err)
+		return cfg
```
Comment (pleshakov) [inline istio/agent/pilot/client.go]: make `GetConfig()` return an error. In this case, you will have `return cfg, fmt.Errorf(...)`

in watcher.go https://github.com/nginxinc/nginmesh/blob/fix-mtls/istio/agent/pilot/watcher.go#L40 do not do deepequal if `w.client.GetConfig()` returns an error.

**Review by pleshakov — CHANGES_REQUESTED** [review]
Please see my comments

**Comment by SirNexus** [comment]
I believe I addressed all of your suggestions. However, I'm wondering about the move of ConfigureAuth() and GetEndpoint() from client.go to main.go. In my mind, both of these functions are for configuring pilot communication and make sense inside of client.go. As well there are a lot of additional packages that are needed to make the functions work in main.go that are already utilized in client.go. Specifically, strings, ioutil and http.

**Review by pleshakov — APPROVED** [review]

**Pull request merged**